NSS-Day-Cohort-53/tabloid-full-stack-always-picturesque-icebergs
# Added Viewing All Published Posts

Resolves #45 

#### Changes Made
1. Added file `IPostRepository` and `PostRepository` to `Repositories` directory.
2. Added file `PostController` to `Controller` directory.
3. Added file `PostManager` to `modules` directory.
4. Added file `PostList` and `Post` to `components` directory.
1. Modified file `IPostRepository` and `PostRepository` to include `GetAllPublishedPosts`, `NewPostFromReader`, and functionality.
2. Modified file `PostController` to include `Get` and functionality.
3. Modified file `PostManager` to include `getAllPublishedPosts` and functionality.
4. Modified file `PostList` to include `PostList` and functionality.
5. Modified file `Post` to include `Post` and functionality.
6. Modified file `Header` to include a nav-link to `PostList`.
7. Modified file `ApplicationViews` to include a route to `PostList`.

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout fs-viewposts
    ```
8. Open a new Terminal tab (⌘T) and navigate to the server directory.
9. Test app functionality.
    > When user logs in AND clicks on the Posts option in the navbar, THEN a list of all posts should appear with the most recent showing first.
10. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.
